src/cursor.rs

```diff
@@ -0,0 +1,163 @@
+use core::fmt;
+use std::error::Error;
+
+use cursor_icon::CursorIcon;
+
+use crate::{icon::PIXEL_SIZE, platform_impl::PlatformCustomCursor};
+
+#[derive(Debug, Clone)]
+pub struct CustomCursor {
+    pub(crate) inner: PlatformCustomCursor,
+}
+
+impl CustomCursor {
+    pub fn from_image(image: CursorImage) -> Self {
```

Comment:
Could we not just implement `fn from_rgba()` here directly?
No reason two introduce multiple types if it's not necessary.

Reply:
You should probably hold off on doing anything here as well if we want to use this as proposed in https://github.com/rust-windowing/winit/pull/3218#discussion_r1395537962.